src/app/game-state/inventory.service.ts

```diff
@@ -1053,7 +1053,14 @@ export class InventoryService {
                 this.itemStacks[mergeDestinationIndex] = null;
                 this.itemStacks[i] = null;
                 this.mergeEquipment(destinationItem, sourceItem, mergeDestinationIndex );
-                return;
+                item = this.itemStacks[mergeDestinationIndex]?.item;
+                if (item){
```

Comment:
Do we need these checks? Seems like we could just skip to setting destinationItem=item.

Reply:
It's part of best practices since there's nothing to ensure that the destination hasn't been nulled during the mergeEquipment call.